muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/coursepicker/CoursePickerRESTService.java

```diff
@@ -375,10 +384,18 @@ public Response listWorkspaces(
                   if (educationType != null) {
                     educationTypeName = educationType.getName();
                   }
+                  
+                  Workspace workspace = workspaceController.findWorkspace(workspaceEntity);
```

Comment:
Fetching `educationSubtypeIdentifier` causes a call to Pyramus (actually, so does resolving `educationTypeName` a few lines above (because of `courseMetaController.findEducationType`)). If workspaces are searched with default max results of 50, a single call to this endpoint causes 100 calls to Pyramus just because of those two pieces of information.

It would be much better if both `educationTypeName` and `mandatority` were indexed to Elastic, from where they could be pulled instantly. The most logical place to do this would probably be `WorkspaceIndexer.indexWorkspace(Workspace workspace, WorkspaceEntity workspaceEntity)`.